### DISCUSSION TITLE
Gelöste Fehler beim Build - aber wiseo überhaupt

### DISCUSSION THREAD

**Discussion by TheMaskedDeveloper** [frickelzeugs/FingerprintDoorbell#18] [Q&A]
Hallo,
erstmal: tolle Idee - wird ungesetzt!
Ich hatte zwei Problem beim Build, die ich für mich lösen konnte:

Ich mußte den lib_ldf_mode von deep auf deep+ umstellen (`lib_ldf_mode = deep+`)
Hat den Fehler gelöst:
```
.pio/libdeps/esp32doit-devkit-v1/AsyncElegantOTA/src/AsyncElegantOTA.h:17:24: fatal error: Update.h: No such file or directory
```

Und bei der Platform die Version auf 3.5.0 setzen, damit nicht 5.2.0 angezogen wurde (`platform = espressif32@3.5.0`)
Hat den Fehler gelöst: 
```
c:/users/larsl/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\esp32doit-devkit-v1\lib239\libESP Async WebServer.a(WebAuthentication.cpp.o):(.literal._ZL6getMD5PhtPc+0x4): undefined reference to `mbedtls_md5_starts'
c:/users/larsl/.platformio/packages/toolchain-xtensa-esp32/bin/../lib/gcc/xtensa-esp32-elf/8.4.0/../../../../xtensa-esp32-elf/bin/ld.exe: .pio\build\esp32doit-devkit-v1\lib239\libESP Async WebServer.a(WebAuthentication.cpp.o): in function `getMD5(unsigned char*, unsigned short, char*)':
C:\Users\larsl\FingerprintDoorbell\FingerprintDoorbell/.pio/libdeps/esp32doit-devkit-v1/ESP Async WebServer/src/WebAuthentication.cpp:73: undefined reference to `mbedtls_md5_starts'
```

Mir ist nur nicht klar, wie es bei mir zu den Fehler kommt. Welche Versionen laufen bei euch?

**Comment by frickelzeugs** [comment]
Danke für deine Info! Das mit dem lib_ldf_mode finde ich interessant, gefällt mir besser als die Lösung irgendeine inoffizielle Lib-Version zu verwenden.

Wie es dazu kam? Nun, die von dir beschriebenen Fehler bekommen mitlerweile alle, die eine frische Installation von PlatformUI mit aktuellen Pakaten verwenden. Als ich irgendwann im ~April das Projekt veröffentlicht habe, konnte ich noch bauen, nachdem ich bei mir nun alles geupdatet habe, bekam ich die selben Fehler. Scheinbar gab es breaking Changes in der Plattform Version 4.0 für irgendeine md5 function, welche die Webserver und OTA-Libraries verwenden. Die beiden Libraries müssen erst gefixt werden, bevor ich auf die aktuellste Plattform-Version hochgehe.
